leapp/cli/upgrade/__init__.py

```diff
@@ -175,6 +175,43 @@ def upgrade(args):
         sys.exit(1)
 
 
+@command('preupgrade', help='Generate preupgrade report')
+@command_opt('--whitelist-experimental', action='append', metavar='ActorName',
+             help='Enables experimental actors')
+def preupgrade(args):
+    if os.getuid():
+        raise CommandError('This command has to be run under the root user.')
+
+    context = str(uuid.uuid4())
+    configuration = {
+        'debug': os.getenv('LEAPP_DEBUG', '0'),
+        'verbose': os.getenv('LEAPP_VERBOSE', '0'),
+        'whitelist_experimental': args.whitelist_experimental or ()
+    }
+    e = Execution(context=context, kind='preupgrade', configuration=configuration)
+    e.store()
+    archive_logfiles()
+    logger = configure_logger('leapp-preupgrade.log')
+    os.environ['LEAPP_EXECUTION_ID'] = context
+
+    try:
+        repositories = load_repositories()
+    except LeappError as exc:
+        sys.stderr.write(exc.message)
+        sys.stderr.write('\n')
+        raise CommandError(exc.message)
+    workflow = repositories.lookup_workflow('IPUWorkflow')()
+    with beautify_actor_exception():
```

Comment:
Beware that the lookup may return None if leapp-repository is broken or not installed. The latter however shouldn't be the case though